File: challenge-16/subset.py
# Python3 program to input 2 sets or lists and check
# whether the 2nd list is a subset of the 1st
# Input Constraint: list_size >= 2

# __main__ script

size_1, size_2 = 0, 0

size_1 = int(input('Input array 1 size: '))

size_2 = int(input('Input array 2 size: '))

# Input Constraint Violated
if size_1 < 2 or size_2 < 2:
	raise Exception('Min Array Size: 2')

print('Input array 1:')

set_1 = set()
set_2 = set()

# Read array 1 and store in set_1
for i in range(size_1):
	number = int(input())
	set_1.add(number)

# Read array 2 and store in set_2
print('\nInput array 2:')

for i in range(size_2):
	number = int(input())
	set_2.add(number)

if set_2.issubset(set_1):
	print('2nd array is a subset of the first')
else:
	print('2nd array is not a subset of the first')

File: challenge-16/Subset.java
/*
* Java program to input 2 arrays
* and check whether the 2nd is a subset of the
* first
* Input Constraint: size >= 2
*/

import java.util.Scanner;
import java.util.Set;
import java.util.HashSet;

public class Subset {

	// Driver code
	public static void main(String[] args) {
		
		// Scanner object for input
		Scanner scanner = new Scanner(System.in);

		int size1, size2;

		System.out.print("Input array 1 size: ");
		size1 = scanner.nextInt();

		System.out.print("Input array 2 size: ");
		size2 = scanner.nextInt();

		// Exit if input cosntraint violated
		if (size1 < 2 || size2 < 2) {
			System.out.println("Error: Min Size 2");
			System.exit(1);
		}

		Set<Integer> set1 = new HashSet<Integer>();
		Set<Integer> set2 = new HashSet<Integer>();

		System.out.println("Input array 1:");

		// Input 1st array and store in HashSet
		for (int i = 0; i < size1; i++) {
			set1.add(scanner.nextInt());
		}

		System.out.println("\nInput array 2:");

		// Input 2nd array and store in HashSet
		for (int i = 0; i < size2; i++) {
			set2.add(scanner.nextInt());
		}

		// If set1 contains all elements of set 2
		if (set1.containsAll(set2)) {
			System.out.println("2nd array is a subset of the first");
		} else {
			System.out.println("2nd array is not a subset of the fist");
		}

	}
}